google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/AppEngineFlexibleSupportProvider.java

```diff
@@ -103,14 +113,40 @@ public void addSupport(@NotNull Module module, @NotNull ModifiableRootModel root
       AppEngineFlexibleFacet facet = FacetManager.getInstance(module).addFacet(
           facetType, facetType.getPresentableName(), null /* underlyingFacet */);
 
-      // Allows suggesting app.yaml and Dockerfile locations in facet and deployment UIs.
       VirtualFile[] contentRoots = rootModel.getContentRoots();
       AppEngineProjectService appEngineProjectService = AppEngineProjectService.getInstance();
+
       if (contentRoots.length > 0) {
-        facet.getConfiguration().setAppYamlPath(
+        Path appYamlPath = Paths.get(
             appEngineProjectService.getDefaultAppYamlPath(contentRoots[0].getPath()));
+
+        // Allows suggesting app.yaml and Dockerfile locations in facet and deployment UIs.
+        facet.getConfiguration().setAppYamlPath(appYamlPath.toString());
         facet.getConfiguration().setDockerfilePath(
             appEngineProjectService.getDefaultDockerfilePath(contentRoots[0].getPath()));
+
+        if (generateConfigurationFilesCheckBox.isSelected()) {
+          if (Files.exists(appYamlPath)) {
```

Comment:
for only 2 conditions like this my preference is combine with && instead of nested if's.